### ISSUE TITLE
Search by shelf / Show self associations in results

### ISSUE THREAD

**Issue by peelusmaximus** [BookStackApp/BookStack#4773]
### Describe the feature you'd like

It would be useful to have the ability to search by shelf. In our use case, we don't use the 1 book / multiple shelves feature. The shelves themselves act as a logical container for information. For further context, we're using bookstack to host I.T documentation. We may have a shelf which has books relating to a hosted environment and another for internal. This isn't something we can achieve using permissions as the same engineers are responsible for both environments.

When searching, there is no indicator currently as to which shelve(s) a result is attributed to. Key word searches are likely to be common across environments but there isn't an easy way to know which results are related to which environment / shelf.

The ideal, would be to search / filter by shelf but even seeing associated shelves for each result would be helpful.

Not remotely a developer so what I'm about to suggest might be absolute tripe :) but if books exist on multiple shelves, could the link to a shelf be effectively, a 'system tag'. One that the user doesn't control directly (as its applied when put on/removed from a shelf) but can filter on using the existing tag filter functionality? 

TLDR
It would be good to see associated shelves in search results
It would be amazing to filter on shelves in search results

### Describe the benefits this would bring to existing BookStack users

It would make the search results more useful. By showing associated shelves in search results it offers more context to the results shown, making them more useful. Being able to filter on shelves makes it easier still to find information quickly.

### Can the goal of this request already be achieved via other means?

Not really. I've considered the following alternatives

- Permissions. It's possible to manipulate visibility with permissions but in our use case that isn't suitable as engineers need access to multiple shelves.
- Tagging. We could filter on tagging, and tag per shelf, book etc but this isn't easy / feasible to implement as tags aren't inherited. Relying on people to manually add tags for every future page / chapter / book will more than likely result in items being missed.

### Have you searched for an existing open/closed issue?

- [X] I have searched for existing issues and none cover my fundamental request

### How long have you been using BookStack?

Under 3 months

### Additional context

Just wanted to add thanks for such a brilliant tool but more than that, the effort you put into documentation too. As a non Linux admin, the fact I've been able to deploy bookstack, enable SAML, configure certificates etc is testament to how good the knowledge articles and videos are.

**Comment by ssddanbrown** [comment]
Thanks for the kind feedback @peelusmaximus.

>  It would be amazing to filter on shelves in search results

I'd consider this the same fundamental request as #3546, which is still open, so will focus on the other request below.
Although, if you have in-shelf search do you still desire shelves in the search results?

>  It would be good to see associated shelves in search results

This has been previously requested in #4080.
The user in there also shared a potential workaround to display shelves (just for books).

Personally I'm not keen on adding complexity, and another layer of UI noise, by showing shelves on each item by default. Especially since in some situations there could be a lot of shelves in use for each item.
Book are typically going to be the top-level hierarchy element shown through parts of BookStack, beyond just search.
As an extra alternative, you could include common internal/external indicator prefixes to books (`I: `, `E:`, or something like that. Would advise a space before the actual name due to how search works).

**Comment by peelusmaximus** [comment]
If there is an in shelf search feature like there is for books that would be perfect to be honest.I only suggested seeing shelves in results as I  thought that might be easier, but fully appreciate how messy that would get for users who do have books that span multiple shelves.Feel free to close this as you're right, the request is a duplicate of 3456 which I didn't see in my initial search. Thanks again!

**Comment by ssddanbrown** [comment]
Okay, thanks for the response.
I'll therefore close this off in favour of existing issue #3546.
